azext_iot/operations/digitaltwin.py

```diff
@@ -0,0 +1,275 @@
+    device_default_interface = _iot_digitaltwin_interface_show(cmd, device_id, INTERFACE_KEY_NAME,
+                                                               hub_name, resource_group_name, login)
+    result = _get_device_default_interface_dict(device_default_interface)
+    return {'interfaces': result}
+
+
+def iot_digitaltwin_command_list(cmd, device_id, source_model, interface=None, schema=False,
+                                 repo_endpoint=PNP_ENDPOINT, repo_id=None, repo_login=None,
+                                 hub_name=None, resource_group_name=None, login=None):
+    result = []
+    target_interfaces = []
+    device_interfaces = _iot_digitaltwin_interface_list(cmd, device_id, hub_name, resource_group_name, login)
+    interface_list = _get_device_default_interface_dict(device_interfaces)
+    interface_count = len([k for k in interface_list if k['name'] == interface])
+    if interface and interface_count == 0:
+        raise CLIError('Target interface is not implemented by the device!')
+
+    if interface:
+        target_interfaces.extend([item for item in interface_list if item['name'] == interface])
+    else:
+        target_interfaces = interface_list
+
+    for entity in target_interfaces:
+        interface_result = {'name': entity['name'], 'urn_id': entity['urn_id'], 'commands': {}}
+        interface_commands = []
+        found_commands = []
+        if source_model.lower() == ModelSourceType.device.value.lower():
+            found_commands = _device_interface_elements(cmd, device_id, entity['urn_id'], INTERFACE_COMMAND,
+                                                        hub_name, resource_group_name, login)
+        else:
+            if source_model.lower() == ModelSourceType.private.value.lower():
+                _validate_repository(repo_id, repo_login)
+            found_commands = _pnp_interface_elements(cmd, entity['urn_id'], INTERFACE_COMMAND,
+                                                     repo_endpoint, repo_id, repo_login)
+        for command in found_commands:
+            command.pop('@type', None)
+            if schema:
+                interface_commands.append(command)
+            else:
+                interface_commands.append(command.get('name'))
+        interface_result['commands'] = interface_commands
+        result.append(interface_result)
+    return {'interfaces': result}
+
+
+def iot_digitaltwin_properties_list(cmd, device_id, source_model, interface=None, schema=False,
+                                    repo_endpoint=PNP_ENDPOINT, repo_id=None, repo_login=None,
+                                    hub_name=None, resource_group_name=None, login=None):
+    result = []
+    target_interfaces = []
+    device_interfaces = _iot_digitaltwin_interface_list(cmd, device_id, hub_name, resource_group_name, login)
+    interface_list = _get_device_default_interface_dict(device_interfaces)
+    interface_count = len([k for k in interface_list if k['name'] == interface])
```

Comment:
In both `iot_digitaltwin_properties_list` and  `iot_digitaltwin_command_list` looks like there are opportunities for refactoring (similar that was done to `iot_digitaltwin_monitor_events`). Idea being to iterate over `interface_list` once. I am still not sure if multiple interfaces with the same name is a valid use case (differentiating between version?). So based on that, you can adjust accordingly.

Example, which is assuming single interface of a particular name.

```python
interface_list = _get_device_default_interface_dict(device_interfaces)
target_interface = next((item for item in interface_list if item['name'] == interface), None)
if interface and not target_interface:
    raise CLIError('Target interface is not implemented by the device!')

target_interfaces.append(target_interface)
...
```

Example, assuming there can be multiple interfaces on a digital twin with the same name (differentiated by version):

```python
interface_list = _get_device_default_interface_dict(device_interfaces)
target_interfaces = [k for k in interface_list if k['name'] == interface]
if interface and not target_interfaces:
  raise CLIError('Target interface is not implemented by the device!')

# You already have interface_list so no need to extend
if not target_interfaces:
    target_interfaces = interface_list

...
```

Reply:
Expecting only one interface, not version based interfaces as of now. So changed accordingly